src/ILLink.RoslynAnalyzer/ISymbolExtensions.cs

```diff
@@ -70,22 +70,49 @@ internal static bool TryGetOverriddenMember (this ISymbol? symbol, [NotNullWhen
 			return overridenMember != null;
 		}
 
+		public static SymbolDisplayFormat ILLinkTypeDisplayFormat { get; } =
+			new SymbolDisplayFormat (
+				typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces,
+				genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters
+			);
+
+		public static SymbolDisplayFormat ILLinkMemberDisplayFormat { get; } =
+			new SymbolDisplayFormat (
+				typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameOnly,
+				genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+				memberOptions:
+					SymbolDisplayMemberOptions.IncludeParameters |
+					SymbolDisplayMemberOptions.IncludeExplicitInterface,
+				parameterOptions: SymbolDisplayParameterOptions.IncludeType
+			);
+
 		public static string GetDisplayName (this ISymbol symbol)
```

Comment:
I'd love to share more code and tests! I gave it some thought but don't see a great way to share more of the display name code (or even interfaces). Even if we could implement interfaces for foreign types we'd need to specialize the implementation for different `IMetadataTokenProvirder`s since the interface doesn't provide enough information to format a display name. Ideas welcome!